cpp/src/arrow/csv/options.h

```diff
@@ -101,6 +101,9 @@ struct ARROW_EXPORT ConvertOptions {
   /// or null by default)
   /// This option is ignored if `include_columns` is empty.
   bool include_missing_columns = false;
+  /// If true, no columns are decoded (record batches will have a row count and no
+  /// columns). This is useful if you need only a row count.
+  bool skip_decoding = false;
```

Comment:
I'd rather if you exposed a separate API. Is it easily doable?

Reply:
It should be doable, we'll just have to be careful to respect all the same options (skip_rows, etc.). I'll make an attempt.